### PR TITLE
Reorganize setup instructions

### LINKED ISSUE #23: Reorganize instructions on generating a session and executing the scraper

### PR DESCRIPTION
**Pull request by alebg**
Fix #23

- Removed all need for .env files everywhere, adapted README accordingly
- Removed checks for telegram password and phone number, or bot token, all of which can now be passed optionally
- Better explained how to generate sessions, and dockerization
- All steps of the README were tested on a fresh install

### CODE REVIEW

**Comment by Deeptikandhol10** [comment]
@maany, what's the status of this pull request review?

**Comment by Deeptikandhol10** [comment]
@maany, what's the status of this pull request review?

**Comment by ydamit** [comment]
@Deeptikandhol10 , I'm taking over this review!

File: README.md [hunk 1]
```diff
@@ -36,58 +15,118 @@ source .venv/bin/activate
 pip install -r requirements.txt
 ```
 
-2. Start a kernel-planckster instance:
-    - Clone the [kernel-planckster](https://github.com/dream-aim-deliver/kernel-planckster) repo elsewhere
-    - Install the required packages, preferable in its own virtual environment, following the instructions in the README
-    - Run it in dev mode with a object store following the README (e.g., `poetry run dev --storage`), where you'll find the host, port, auth key and schema (they should match this repo's `.env.example` file)
+2. You will need a phone number and a Telegram account. Obtain the following credentials linked to your account, from [Telegram](https://core.telegram.org/api/obtaining_api_id): API ID, and API hash.
+    i. **Note 1**: You will need access to the phone linked to the telegram account to which the API ID and hash belong, as Telegram will send a verification code to it the first time you run the scraper. The phone number must be in international format (e.g., +391234567890 for an Italian number).
```
Comment (ydamit) [inline README.md:19]:
```suggestion
    i. **Note 1**: You will need access to the phone linked to the Telegram account to which the API ID and hash belong, as Telegram will send a verification code to it the first time you run the scraper. The phone number must be in international format (e.g. +391234567890 for an Italian number).
```

File: README.md [hunk 2]
```diff
@@ -36,58 +15,118 @@ source .venv/bin/activate
...
+3. Obtain an OpenAI API key from the [OpenAI website](https://openai.com/api/) and take note of it
 
-4. Copy the `.env.example` file to `.env` and fill in the required fields.
-    - For `KERNEL_PLANCKSTER_*`, get them from the kernel-planckster README or the instance you started in step 3
-    - You can choose the `STORAGE_PROTOCOL` to use:
-        + `s3`
-        + `local` for local storage, will create a `data` directory in the root of the project and store the files there. DEPRECATED: use `s3` with Kernel Planckster running on the side
-    - The `TELEGRAM_*` fields are the credentials you obtained in step 4
+4. Setup kernel-planckster locally:
+    1. Clone the [kernel-planckster](https://github.com/dream-aim-deliver/kernel-planckster) repo elsewhere
```
Comment (ydamit) [inline README.md:25]:
```suggestion
    1. Clone the [kernel-planckster](https://github.com/dream-aim-deliver/kernel-planckster) repo
```

File: README.md [hunk 3]
```diff
@@ -36,58 +15,118 @@ source .venv/bin/activate
...
+    2. Install the required packages, preferable in its own virtual environment, following the instructions in the README
```
Comment (ydamit) [inline README.md:26]:
```suggestion
    2. Install the required packages, preferably in a virtual environment, following the instructions in the kernel-planckster README
```

File: README.md [hunk 4]
```diff
@@ -36,58 +15,118 @@ source .venv/bin/activate
...
+    3. Run it in dev mode with a object store following the README (e.g., `poetry run dev:storage`), where you'll find the host, port, auth key and schema
```
Comment (ydamit) [inline README.md:27]:
```suggestion
    3. Run Kernel Planckster in dev mode with a object store following the README (e.g., `poetry run dev:storage`), where you'll find the host, port, auth key and schema
```

File: README.md [hunk 5]
```diff
@@ -36,58 +15,118 @@ source .venv/bin/activate
...
+This .session file can be shared with other team members to avoid having to re-enter the verification code on their machines.
+
+#### Minimizing Prompting with Optional Command line arguments
 
+The following optional command line arguments can be passed to the main scraper script to minimize the need for manual intervention the first time the scraper is run:
 
-### Build Image
+- `--telegram-phone-number`: The phone number linked to the telegram account. Must be in international format (e.g., +391234567890 for an Italian number).
+- `--telegram-password`: The password of the telegram account, if it is password protected.
+- `--telegram-bot-token`: The token of the telegram bot that will be used to send messages to the channel.
```
Comment (ydamit) [inline README.md:60]:
```suggestion
- `--telegram-phone-number`: The phone number linked to the Telegram account. Must be in international format (e.g., +391234567890 for an Italian number).
- `--telegram-password`: The password of the Telegram account, if it is password protected.
- `--telegram-bot-token`: The token of the Telegram bot, if applicable, that will be used to send messages to the channel.
```

**Review by ydamit — APPROVED** [review]
Looks good! All my comments are purely spelling/stylistic changes in the Readme. No need to hold up merging for that, but might as well record my suggestions somewhere.

**Review by maany — APPROVED** [review]

**Pull request merged**